src/commands/base.mjs

```diff
@@ -61,5 +61,93 @@ export class BaseCommand extends ShellRunner {
     this.chartManager = opts.chartManager
     this.configManager = opts.configManager
     this.depManager = opts.depManager
+    this._configMaps = new Map()
+  }
+
+  /**
+   * Dynamically builds a class with properties from the provided list of flags
+   * and extra properties, will keep track of which properties are used.  Call
+   * getUnusedConfigs() to get an array of unused properties.
+   *
+   * @param {string} configName the name of the configuration
+   * @param {CommandFlag[]} flags an array of flags
+   * @param {string[]} [extraProperties] an array of extra properties
+   * @returns {Object} the instance of the new class
+   */
+  getConfig (configName, flags, extraProperties = []) {
```

Comment:
FYI, Nathan and I reviewed this in a 1:1.  He is okay with how I'm using this here and what I'm using it for.  It is more of an exception though, and not necessarily to be used as a pattern.

Reply:
I am most likely missing something here as I don't understand the need to have this dynamic class generation and if I ever look at or touch this code, most likely I shall be tempted to refactor.

If we need a class to track the used vs unused flags, we could just have a config class with the full set of configs (so a centralized management of all flags), then the constructor would take a list of flags and mark the corresponding flags as used (or mark the other ones as unused), and then `getUnusedConfigs` would return the unused flags.

However, if two of you are ok with this, then let's go with it.

Reply:
@leninmehedy , thank you for the review and feedback.  I do like what you suggested.  I think I was also trying to have the goal of having the IDE provide tool tips.  Like this:
![image](https://github.com/user-attachments/assets/46653569-afa4-4a63-b3aa-5a3c81d5a228)
Which also requires the JSDoc to work correctly.  However, you lose the IDE tool tips here:
![image](https://github.com/user-attachments/assets/5d572397-df5b-4b25-a6d5-2cc3ec5a47da)
Unless you do something like this:
![image](https://github.com/user-attachments/assets/0eec7d15-d192-4045-bf94-e6eee24f11bc)